src/experimental-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -166,26 +188,30 @@ export default class MeshLayer extends Layer {
   }
 
   getModel(gl) {
-    const isValidMesh = this.props.mesh instanceof Geometry && this.props.mesh.attributes.positions;
-    assert(isValidMesh);
-
     return new Model(
       gl,
       Object.assign({}, this.getShaders(), {
         id: this.props.id,
-        geometry: this.props.mesh,
+        geometry: getGeometry(this.props.mesh),
         isInstanced: true
       })
     );
   }
 
-  loadTexture(src) {
+  setTexture(src) {
     const {gl} = this.context;
-    const {model} = this.state;
-    getTexture(gl, src).then(texture => {
-      model.setUniforms({sampler1: texture});
-      this.setNeedsRedraw();
-    });
+    const {model, emptyTexture} = this.state;
+
+    if (src) {
+      getTexture(gl, src).then(texture => {
+        model.setUniforms({sampler: texture, hasTexture: 1});
+        this.setState({texture});
+      });
+    } else {
+      // reset
+      this.state.model.setUniforms({sampler: emptyTexture, hasTexture: 0});
```

Comment:
Why do we need an empty texture if we have a `hasTexture` flag? Assume it is to work around the uniform checks in luma.gl?

Reply:
Yes. Is there a way to disable the warning?

Reply:
No. But maybe there should be. Not clear how to do it though.
* A special reserved uniform key `ignoreWarnings`? 
* An extra map `unusedAttributes`?
* A special value: `UNUSED`?

Althernatively, I seem to remember THREE.js adds a dummy texture (a simple black and white grid, probably like 2x2 pixels programmatically created) when a real texture is not supplied. This could of course be combined with the above.